src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/PooledTxsRequestor.cs

```diff
@@ -15,17 +15,33 @@
 
 namespace Nethermind.Network.P2P.Subprotocols.Eth
 {
-    public class PooledTxsRequestor(ITxPool txPool, ITxPoolConfig txPoolConfig, ISpecProvider specProvider) : IPooledTxsRequestor
+    public class PooledTxsRequestor : IPooledTxsRequestor, IDisposable
     {
         private const int MaxNumberOfTxsInOneMsg = 256;
-        private readonly bool _blobSupportEnabled = txPoolConfig.BlobsSupport.IsEnabled();
-        private readonly long _configuredMaxTxSize = txPoolConfig.MaxTxSize ?? long.MaxValue;
+        private readonly bool _blobSupportEnabled;
+        private readonly long _configuredMaxTxSize;
 
-        private readonly long _configuredMaxBlobTxSize = txPoolConfig.MaxBlobTxSize is null
+        private readonly long _configuredMaxBlobTxSize;
+
+        private readonly ClockKeyCache<ValueHash256> _pendingHashes = new(MemoryAllowance.RequestedTxHashCacheSize);
```

Comment:
size reduced from 131k to just 4k?

Reply:
Is twice the size of mempool; and we are now clearing each block

On the request side we check if what we are requesting is "known" by tx pool; so this size and not clearing is both over cautious and not under control of txPool to control re-requesting (if it kicks out a tx)

What this currently could mean is we request a tx; peer doesn't give it to us, so we never request it again until we have received 131_072 different txs

The change is so we won't request if it "known" by txpool or (is in last 2048 txs requested and same head block)